pkg/integrations/v4/logs/cfg_test.go

```diff
@@ -1813,3 +1813,247 @@ func TestDetermineUseAnsiFlagValue(t *testing.T) {
 		})
 	}
 }
+
+func TestGetTotalTargetFilesForPath(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name          string
+		logCfg        LogCfg
+		expectedCount int
+	}{
+		{
+			"File path log config",
+			LogCfg{
+				Name: "log-file-1",
+				File: "./*_test.go", // we count the number of test files in /pkg/integrations/v4/logs/ folder
+			},
+			4,
+		},
+		{
+			"Non file path log config - Syslog",
+			LogCfg{
+				Name: "syslog-tcp-test",
+				Syslog: &LogSyslogCfg{
+					URI:    "tcp://0.0.0.0:5140",
+					Parser: "syslog-rfc5424",
+				},
+			},
+			0,
+		},
+	}
+
+	for _, testItem := range tests {
+		// Prevent the loop variable from being captured in the closure below
+		test := testItem
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			actualCount := getTotalTargetFilesForPath(test.logCfg)
+			assert.Equal(t, actualCount, test.expectedCount)
+		})
+	}
+}
+
+func TestNewFbConfForTargetFileCount(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name          string
+		logFwd        config.LogForward
+		logsCfg       LogsCfg
+		expectedCount int
+	}{
+		{
+			"Target File Count test for File path config",
+			logFwdCfg,
+			LogsCfg{
+				{
+					Name: "log-file-1",
+					File: "./*_test.go", // we count the number of test files in /pkg/integrations/v4/logs/ folder
+				},
+			},
+			4,
+		},
+		{
+			"Target File Count test-2 for File path config",
+			logFwdCfg,
+			LogsCfg{
+				{
+					Name: "log-file-1",
+					File: "./*.go", // we count the number of go files in /pkg/integrations/v4/logs/ folder
+				},
+			},
+			11,
+		},
+		{
+			"Target File Count test-2 for Non-file path config",
+			logFwdCfg,
+			LogsCfg{
+				{
+					Name: "syslog-tcp-test",
+					Syslog: &LogSyslogCfg{
+						URI:    "tcp://0.0.0.0:5140",
+						Parser: "syslog-rfc5424",
+					},
+				},
+			},
+			0,
+		},
+	}
+
+	for _, testItem := range tests {
+		// Prevent the loop variable from being captured in the closure below
+		test := testItem
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			NewFBConf(test.logsCfg, &test.logFwd, "0", "")
+			assert.Equal(t, test.expectedCount, test.logsCfg[0].TargetFilesCnt)
+		})
+	}
+}
+func TestTooManyFilesWarningFormat(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name              string
+		warningDetails    TooManyFilesWarning
+		expectedOutputMsg string
+	}{
+		{
+			"Too Many files with multiple file paths",
+			TooManyFilesWarning{
+				1500,
+				LogsCfg{
+					{
+						Name:           "log-file-1",
+						File:           "file.path-1",
+						TargetFilesCnt: 1000,
+					},
+					{
+						Name:           "log-file-2",
+						File:           "file.path-2",
+						TargetFilesCnt: 500,
+					},
```

Comment:
Please also include a path with 0 files to stress what I mentioned in `cfg_template.go`:
```

{
    Name:           "log-file-3",
    File:           "file.path-3",
    TargetFilesCnt: 0,
}
```